### PR TITLE
Fix perf issues in allocation and make bootstrappers sequential

### PR DESCRIPTION
**Pull request by robskillington**
Bootstrapping in parallel adds a ton of extra pressure in scheduling goroutines and using resources, this change makes existing bootstrappers sequential.  This shows a marked improvement both in write capability and also too with the duration length to bootstrap overall.  

The fs bootstrapper only uses a single goroutine to open references to file sets now which means there is less fs reader creation which helps avoid creating many large index buffers during bootstrapping and only opens the fs readers just before use by the processors.

The concurrency for peer streaming is also set to half CPUs by default instead of 1024 which just created too many concurrent streams to manage.

There is also some allocations that are heavy that don't need to be, i.e. no bytes buffering in the simple identifier pool, not using a proto buffer in the fs reader and a few other places.

cc @xichen2020 @kobolog @prateek

### CODE REVIEW

File: client/session.go [hunk 1]
```diff
@@ -2334,7 +2315,7 @@ type enqueueChannel struct {
 
 func newEnqueueChannel(m *streamFromPeersMetrics) *enqueueChannel {
 	c := &enqueueChannel{
-		peersMetadataCh: make(chan []*blocksMetadata, 4096),
+		peersMetadataCh: make(chan []*blocksMetadata, 32768),
```
Comment (xichen2020) [inline client/session.go]: Better to make this a const?

File: persist/fs/read.go [hunk 2]
```diff
@@ -82,6 +82,7 @@ type reader struct {
 	entriesRead int
 	indexUnread []byte
 	prologue    []byte
+	protoBuff   *proto.Buffer
```
Comment (xichen2020) [inline persist/fs/read.go]: `protoBuf` might be more consistent with `unreadBuf` and `digestBuf`?

File: storage/bootstrap/bootstrapper/peers/source.go [hunk 3]
```diff
@@ -78,33 +78,41 @@ func (s *peersSource) Read(
...
 	for shard, ranges := range shardsTimeRanges {
-		it := ranges.Iter()
-		for it.Next() {
-			currRange := it.Value()
-			// We fetch all results in parallel and the underlying session will
-			// rate limit the total throughput with a single worker pool that
-			// is defined by the fetch series blocks batch concurrency option
-			wg.Add(1)
-			go func(shard uint32, start, end time.Time) {
+		shard, ranges := shard, ranges
```
Comment (prateek) [inline storage/bootstrap/bootstrapper/peers/source.go]: nit: unnecessary var assignment?

File: client/session.go (hunk 1 above)
Comment (robskillington) [inline client/session.go]: Sure thing.

File: persist/fs/read.go (hunk 2 above)
Comment (robskillington) [inline persist/fs/read.go]: Sure thing.

File: storage/bootstrap/bootstrapper/peers/source.go (hunk 3 above)
Comment (robskillington) [inline storage/bootstrap/bootstrapper/peers/source.go]: Yeah, should remove this now that we don't need to capture it for the goroutine that used to encapsulate the inner logic.

Comment (robskillington) [inline storage/bootstrap/bootstrapper/peers/source.go]: Done.

**Pull request merged**